### PR TITLE
add automatic reconnect websocket and janus session on janus server r…

### PR DESCRIPTION
**Pull request by arpu**
…estart

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -87,13 +88,21 @@ class JanusAdapter {
...
-    this.ws = new WebSocket(this.serverUrl, "janus-protocol");
-    this.session = new mj.JanusSession(this.ws.send.bind(this.ws));
-    this.ws.addEventListener("open", this.onWebsocketOpen);
-    this.ws.addEventListener("message", this.onWebsocketMessage);
+    this.ws = new WebSocketClient(this.serverUrl, "janus-protocol", {
+      strategy: 'fibonacci',
+      randomisationFactor: 0, // defaults to 0, must be between 0 and 1
+      initialDelay: 1,       // defaults to 100 ms
+      maxDelay: 300           // defaults to 10000 ms
+    });
```
Comment (mquander) [inline src/index.js]: Where did you learn about these parameters? If I go look at the source in https://github.com/gamestdio/websocket/blob/master/src/backoff.js I see only `initialDelay` -- I don't see `randomisationFactor` or `maxDelay`.

Maybe this should just be
```Javascript
this.ws = new WebSocketClient(this.serverUrl, "janus-protocol", {
  backoff: 'fibonacci',
  initialDelay: '1'
});
```
?

**Comment by mquander** [comment]
Thanks for considering to do this! This seems like an improvement and I don't mind taking this as a dependency -- @robertlong thinks it's reasonable also. I'm happy to merge as soon as we nail down the initialization.

File: src/index.js (hunk 1 above)
Comment (arpu) [reply src/index.js]: yeah looks like it was removed in this commit https://github.com/gamestdio/websocket/commit/717a90b13c86bea55e7e315829e05c0dd5416934

i will update the PR

File: src/index.js [hunk 2]
```diff
@@ -87,13 +88,19 @@ class JanusAdapter {
 
   connect() {
     debug(`connecting to ${this.serverUrl}`);
-    this.ws = new WebSocket(this.serverUrl, "janus-protocol");
-    this.session = new mj.JanusSession(this.ws.send.bind(this.ws));
-    this.ws.addEventListener("open", this.onWebsocketOpen);
-    this.ws.addEventListener("message", this.onWebsocketMessage);
+    this.ws = new WebSocketClient(this.serverUrl, "janus-protocol", {
+      strategy: 'fibonacci',
```
Comment (mquander) [inline src/index.js]: I still don't think this is quite right -- the code [here](https://github.com/gamestdio/websocket/blob/master/src/index.js#L16) is looking for the strategy parameter to be named `backoff`, not `strategy`.

Incidentally, I have no particular opinion about what the backoff strategy should be. I would be interested if @robertlong or @InfiniteLee thought they had an opinion about it.

**Comment by arpu** [comment]
i will delete this PR and add a new one 

(i delete my fork by accident )

**Pull request closed without merging**